### PR TITLE
FEATURES:  Localisation support, Custom Wizard plugin integration, code improvements, static data moved to yml files

### PR DESCRIPTION
**Pull request by merefield**
Added:
- Localisation support
- Custom Wizard plugin integration
- Code improvements
- Static data moved into yml files

### CODE REVIEW

File: assets/javascripts/discourse/components/user-nationalflags-preferences.js.es6 [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { ajax } from 'discourse/lib/ajax';
+
+export default Ember.Component.extend({
+  layoutName: 'javascripts/discourse/templates/connectors/user-custom-preferences/user-nationalflags-preferences',
+  natflaglist: [],
+
+  @on('init')
+  setup() {
+
+    ajax({
```
Comment (angusmcleod) [inline assets/javascripts/discourse/components/user-nationalflags-preferences.js.es6]: @merefield This is fine, but the more natural place for this ajax call is in the ``afterModel`` hook in a user-preferences route, added via the pluginApi in an initializer, then adding the data to the model which is passed to this plugin outlet, e.g.

```
      api.modifyClass('route:preferences', {
        afterModel(model) {
          return ajax('/natflags/flags').then(result => {
             localisedFlags = natflags.flags.map (element => {
                return {  code: element.code, 
                  pic: element.pic, 
                  description: I18n.t(`flags.description.${element.code}`)
               }
             });
            model.set('localisedFlags', localisedFlags);
          })
        }
      });
```

File: plugin.rb [hunk 2]
```diff
@@ -11,6 +11,28 @@
...
+  module ::DiscourseNationalFlags
+    class Engine < ::Rails::Engine
+      engine_name "discourse_national_flags"
+      isolate_namespace DiscourseNationalFlags
+    end
+  end
+
+  load File.expand_path('../lib/flags.rb', __FILE__)
+
+  require_dependency 'about_controller'
```
Comment (angusmcleod) [inline plugin.rb]: @merefield require_dependency is no longer needed after the zeitwork upgrade in core Discourse.

**Review by angusmcleod — COMMENTED** [review]
@merefield Looks good! just two notes from me.

File: assets/javascripts/discourse/components/user-nationalflags-preferences.js.es6 (hunk 1 above)
Comment (merefield) [reply assets/javascripts/discourse/components/user-nationalflags-preferences.js.es6]: OK sure, done.

File: plugin.rb (hunk 2 above)
Comment (merefield) [reply plugin.rb]: Removed, thanks

**Comment by Ebsy** [comment]
Merged pull request

**Pull request merged**